frontend/src/pages/detail/ModfiyBook.tsx

```diff
@@ -35,156 +45,266 @@ export default function BookDetail() {
     detailState.isConfirmationPopupVisible
   );
 
-  const onSubmit = (): void => {
-    setIsModifiable(false);
-  };
+  // Toast is used for showing success/error messages
+  const toast = useRef<Toast>(null);
 
-  // TODO: Fix these set functions
-  const setAuthorsModified = (authors: string[]): void => {
-    setAuthors(authors);
+  const showSuccess = () => {
+    toast.current?.show({ severity: "success", summary: "Book Edited" });
   };
 
-  // TODO: Fix these set functions
-  const setGenresModified = (genres: string[]): void => {
-    setGenres(genres);
+  const showFailure = (message: string) => {
+    toast.current?.show({ severity: "error", summary: message });
   };
 
+  // Validation for the form
+  const formik = useFormik({
+    initialValues: {},
+
+    validate: () => {
+      const errors: FormikErrors<ErrorDisplay> = {};
+
+      if (!genre) {
+        errors.message = "Genre is a required field";
+      }
+
+      if (!price) {
+        errors.message = "Retail Price is a required Field";
+      }
+
+      if (errors.message) {
+        showFailure(errors.message);
+      }
+
+      return errors;
+    },
+    onSubmit: () => {
+      const book: Book = {
```

Comment:
why is this field needed?

Reply:
Needed to follow the book interface for passing data into BOOKS_API.modifyBook. So had to arrange the data in this way.